IBPSA/Fluid/FixedResistances/CheckValve.mo

```diff
@@ -0,0 +1,143 @@
+
+  parameter Real kFixed(unit="", min=0)=
+    if dpFixed_nominal > Modelica.Constants.eps then
+      m_flow_nominal/sqrt(dpFixed_nominal)
+    else 0
+    "Flow coefficient of fixed resistance that may be in series with valve, 
+    k=m_flow/sqrt(dp), with unit=(kg.m)^(1/2).";
+
+  Real k(unit="", min=Modelica.Constants.small)
+    "Flow coefficient of valve and pipe in series in allowed/forward direction, 
+    k=m_flow/sqrt(dp), with unit=(kg.m)^(1/2).";
+
+protected
+  Real a     "Pressure scaled variable";
+  Real cv    "Smoothed heaviside checkvalve characteristic";
+  Real kstar "Smoothed restriction characteristic";
+
+initial equation
+  assert(dpFixed_nominal > -Modelica.Constants.eps,
+    "In " + getInstanceName() + ": Require dpFixed_nominal >= 0. 
+    Received dpFixed_nominal = " + String(dpFixed_nominal) + " Pa.");
+
+  assert(l > -Modelica.Constants.eps,
+    "In " + getInstanceName() + ": Require l >= 0. Received l = " + String(l));
+
+equation
+
+  a = dp/dpValve_nominal;
+  cv = max(0,min(1,a^3*(10+a*(-15+6*a)))); // update this to call
+  //IBPSA.Utilities.Math.Functions.smoothHeaviside when issue 1202 is merged
+  kstar = Kv_SI*(cv + l);
+
+  // add series restriction when applicable
+  if (dpFixed_nominal > Modelica.Constants.eps) then
+    k = sqrt(1/(1/kFixed^2 + 1/kstar^2));
+  else
+    k = kstar;
+  end if;
+
+  if homotopyInitialization then m_flow = homotopy(actual=
+      IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp(
+        dp = dp, k = k, m_flow_turbulent = m_flow_turbulent),
+      simplified = if dp > 0 then k*dp else k*l*dp);
+  else
+    // do not use homotopy
+    m_flow = IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp(
+      dp = dp, k = k, m_flow_turbulent = m_flow_turbulent);
+  end if;
+
+  annotation (Icon(coordinateSystem(preserveAspectRatio=true, extent={{-100,-100},
+            {100,100}}), graphics={
+        Polygon(
+          points={{-100,-100},{-100,100},{100,100},{100,-100}},
+          lineColor={255,255,255},
+          fillColor={255,255,255},
+          fillPattern=FillPattern.Solid),
+        Line(
+          points={{-100,0},{-70,0}},
+          color={0,128,255},
+          lineThickness=0.5),
+        Line(
+          points={{0,70},{-70,0}},
+          color={0,128,255},
+          lineThickness=0.5),
+        Line(
+          points={{0,-70},{-70,0}},
+          color={0,128,255},
+          lineThickness=0.5),
+        Ellipse(
+          extent={{-40,-55},{70,55}},
+          lineColor={0,128,255},
+          fillColor={255,255,255},
+          lineThickness=0.5,
+          fillPattern=FillPattern.Solid),
+        Line(
+          points={{70,0},{100,0}},
+          color={0,128,255},
+          lineThickness=0.5)}), Documentation(info="<html>
+<p>Implementation of a hydraulic check valve. </p>
+<h4>Main equations</h4>
+<p>The basic flow function <code>IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp</code> 

```

Comment:
```suggestion
<p>The basic flow function <a href="modelica://IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp">IBPSA.Fluid.BaseClasses.FlowModels.basicFlowFunction_dp</a> 
```

Links to models are explicitly linked since then the refactoring scripts of buildingspy also update the documentation when a file is renamed.